core/src/main/scala/sttp/tapir/ShadowedEndpoint.scala

```diff
@@ -0,0 +1,17 @@
+package sttp.tapir
+
+import sttp.model.Method
+
+case class ShadowedEndpoint(e: Endpoint[_, _, _, _], by: Endpoint[_, _, _, _]) {
+  override def toString = e.input.show + ", is shadowed by: " + by.input.show
+}
+
+private sealed trait PathComponent
```

Comment:
better, but I'd still move these to the only place where they are used - inside the `ShadowedEndpointChecker` :)